### PR TITLE
Refactor detail view components

### PR DESCRIPTION
**Pull request by caseyszilagyi**
## Feature Description/Implementation

Extracted a bunch of the common creation of components and cleaned up the detail view for PO/SR/BB so they're more readable. Still a lot of work to be done but should make it easier to update the UI

## Dependencies

Detail views for PO/SR/BB

### CODE REVIEW

File: frontend/src/components/OneDayCalendar.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    <div>
+      <label
+        htmlFor="date"
+        className="p-component text-teal-900 p-text-secondary my-auto pr-2"
+      >
+        Date
+      </label>
+      <Calendar
+        id="date"
+        disabled={!props.isModifiable}
```
Comment (nex-pixel) [inline frontend/src/components/OneDayCalendar.tsx:20]: The not is kind of odd. Here would much rather have a disabled field show through with that being optional

File: frontend/src/components/buttons/AddRowButton.tsx [hunk 2]
```diff
@@ -29,13 +31,16 @@ export default function AddRowButton<T extends IDer>(
...
-      disabled={!!props.isModifiable}
-    />
+    <>
+      <Button
+        type="button"
+        label={props.label ?? "Add"}
+        icon="pi pi-plus"
+        className="p-button-info mr-2"
+        onClick={addNewRow}
+        disabled={props.isDisabled}
```
Comment (nex-pixel) [inline frontend/src/components/buttons/AddRowButton.tsx:41]: do we need ?? defaults or no

File: frontend/src/components/buttons/DeleteButton.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { Button } from "primereact/button";
+
+interface DeleteButtonProps {
+  isEnabled: boolean;
+  onClick: () => void;
+}
+
+export default function DeleteButton(props: DeleteButtonProps) {
+  return (
+    <div className="flex col-1">
```
Comment (nex-pixel) [inline frontend/src/components/buttons/DeleteButton.tsx:10]: we don't want to factor out <div className="flex col-1"> this is page specific code that creates margins and padding for each JSX element.

File: frontend/src/components/buttons/EditCancelButton.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import { Button } from "primereact/button";
+
```
Comment (nex-pixel) [inline frontend/src/components/buttons/EditCancelButton.tsx:2]: Rename file include the isAddPage functionality maybe EditCancelButtonforAddDetailPage

File: frontend/src/components/text/AddDetailModifyTitle.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  addTitle: string;
+  modifyTitle: string;
+  detailTitle: string;
+  isAddPage: boolean;
+  isModifyPage: boolean;
+}
+
+export default function DefaultTitleText(props: TitleTextProps) {
+  return (
+    <div className="pt-2 col-10">
```
Comment (nex-pixel) [inline frontend/src/components/text/AddDetailModifyTitle.tsx:11]: again same thing with the <div> stuff from earlier

File: frontend/src/components/text/TotalDollars.tsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import { priceBodyTemplate } from "../../util/TableCellEditFuncs";
+
+interface TotalDollarsProps {
+  label: string;
+  totalDollars: number;
+}
+
+export default function TotalDollars(props: TotalDollarsProps) {
+  return (
+    <div className="flex">
```
Comment (nex-pixel) [inline frontend/src/components/text/TotalDollars.tsx:10]: Ignore this <div> it acts as a modifier to self align the items inside

File: frontend/src/components/uploaders/CSVFileUploader.tsx [hunk 7]
```diff
@@ -4,22 +4,27 @@ const MAX_FILE_UPLOAD = 1000000; // Bytes
...
-      />
-    </div>
+    <>
+      {props.visible && (
+        <div className="card flex justify-content-center">
+          <FileUpload
+            auto
+            mode="basic"
+            accept=".csv"
+            maxFileSize={MAX_FILE_UPLOAD}
```
Comment (nex-pixel) [inline frontend/src/components/uploaders/CSVFileUploader.tsx:20]: This is outside the refactoring comment, but did professor Bletsch say anything about max file size for CSV?

**Review by nex-pixel — CHANGES_REQUESTED** [review]

File: frontend/src/components/buttons/AddRowButton.tsx (hunk 2 above)
Comment (caseyszilagyi) [reply frontend/src/components/buttons/AddRowButton.tsx:41]: We actually don't, but I'm going to add them for clarity

Comment (caseyszilagyi) [reply frontend/src/components/buttons/AddRowButton.tsx:41]: If undefined, it uses the primereact default. But I think it's better to show explicitly what will happen

File: frontend/src/components/uploaders/CSVFileUploader.tsx (hunk 7 above)
Comment (caseyszilagyi) [reply frontend/src/components/uploaders/CSVFileUploader.tsx:20]: Nope, maybe worth asking him in class

**Review by nex-pixel — APPROVED** [review]

**Pull request merged**